game/scripts/vscripts/components/progression/hero_progression.lua

```diff
@@ -155,3 +156,9 @@ function HeroProgression:OnLevelUpChatCmd(keys)
   DebugPrint('Levelling up ' .. hero:GetName() .. ' now at level ' .. hero:GetLevel())
   hero:HeroLevelUp(true)
 end
+
+function HeroProgression:ExperienceFilter(keys)
+  local playerID = keys.player_id_const
+
+  return PlayerResource:GetConnectionState(playerID) ~= DOTA_CONNECTION_STATE_DISCONNECTED
```

Comment:
Should have looked at this sooner, but anyway, I think this should be `== DOTA_CONNECTION_STATE_CONNECTED` because players could also be abandoned or some other state.